### PR TITLE
Replace custom math lib with modules from QT

### PR DESCRIPTION
**Pull request by Eng-MohamedHussien**
- Delete all headers from math lib with their corresponding tests
- Use QVector2D & QVector4D instead of Vector2 & Vector4
- Update mocks to be compatible with QVector2D & QVector4D

Ticket: SOUR-63

### CODE REVIEW

File: tests/integration/lib/CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,7 @@ foreach(test_name IN LISTS test_lib_names)
     PUBLIC GTest::gtest
            GTest::gmock
            Qt5::Core
+           Qt5::Gui
```
Comment (wow2006) [inline tests/integration/lib/CMakeLists.txt:18]: You removed all tests why you added the gui library?

File: src/messaging/type/graph/MessageGraphNodeMove.h [hunk 2]
```diff
@@ -16,9 +18,15 @@ class MessageGraphNodeMove final : public Message<MessageGraphNodeMove> {
   }
 
   void print(std::wostream& ostream) const override {
-    ostream << tokenId << L" " << delta.toWString();
+    const auto toWString = [this]() {
+      std::wstringstream result;
+      result << L"[" << delta.x() << L", " << delta.y() << L"]";
+
+      return result.str();
```
Comment (wow2006) [inline src/messaging/type/graph/MessageGraphNodeMove.h]:
```suggestion
      return fmt::format(L"[{}, {}]", delta.x(), delta.x());
```

File: src/lib_gui/qt/graphics/graph/QtGraphNode.cpp [hunk 3]
```diff
@@ -95,56 +95,56 @@ const std::list<QtGraphNode*>& QtGraphNode::getSubNodes() const {
...
-  Vec2i currentPosition = getPosition();
-  Vec2i offset = position - currentPosition;
+bool QtGraphNode::setPosition(const QVector2D& position) {
+  QVector2D currentPosition = getPosition();
+  QVector2D offset = position - currentPosition;
 
-  if(offset.x != 0 || offset.y != 0) {
-    this->moveBy(offset.x, offset.y);
-    setColumnSize(Vec2i());
+  if(offset.x() != 0 || offset.y() != 0) {
```
Comment (wow2006) [inline src/lib_gui/qt/graphics/graph/QtGraphNode.cpp]: comparing float with zero is not a good idea. Checl the following https://www.fitzgibbon.ie/floating-point-equality

File: src/lib_gui/qt/graphics/graph/QtGraphNode.cpp [hunk 4]
```diff
@@ -95,56 +95,56 @@ const std::list<QtGraphNode*>& QtGraphNode::getSubNodes() const {
...
+    this->moveBy(offset.x(), offset.y());
+    setColumnSize(QVector2D());
```
Comment (wow2006) [inline src/lib_gui/qt/graphics/graph/QtGraphNode.cpp]:
```suggestion
    setColumnSize({});
```

File: src/lib_gui/qt/graphics/graph/QtGraphNode.cpp [hunk 5]
```diff
@@ -95,56 +95,56 @@ const std::list<QtGraphNode*>& QtGraphNode::getSubNodes() const {
...
 }
 
 QSize QtGraphNode::size() const {
-  return QSize(m_size.x, m_size.y);
+  return QSize(m_size.x(), m_size.y());
 }
 
 void QtGraphNode::setSize(QSize size) {
-  setSize(Vec2i(size.width(), size.height()));
+  setSize(QVector2D(size.width(), size.height()));
```
Comment (wow2006) [inline src/lib_gui/qt/graphics/graph/QtGraphNode.cpp]:
```suggestion
  setSize({size.width(), size.height()});
```

File: src/lib_gui/qt/graphics/graph/QtGraphNode.cpp [hunk 6]
```diff
@@ -95,56 +95,56 @@ const std::list<QtGraphNode*>& QtGraphNode::getSubNodes() const {
...
 }
 
-Vec4i QtGraphNode::getBoundingRect() const {
-  Vec2i pos = getPosition();
-  Vec2i size = getSize();
-  return Vec4i(pos.x, pos.y, pos.x + size.x, pos.y + size.y);
+QVector4D QtGraphNode::getBoundingRect() const {
+  QVector2D pos = getPosition();
+  QVector2D size = getSize();
+  return QVector4D(pos.x(), pos.y(), pos.x() + size.x(), pos.y() + size.y());
```
Comment (wow2006) [inline src/lib_gui/qt/graphics/graph/QtGraphNode.cpp]:
```suggestion
  return {pos.x(), pos.y(), pos.x() + size.x(), pos.y() + size.y()};
```

**Review by wow2006 — CHANGES_REQUESTED** [review]

File: src/lib_gui/qt/graphics/graph/QtGraphNode.cpp (hunk 6 above)
Comment (Eng-MohamedHussien) [reply src/lib_gui/qt/graphics/graph/QtGraphNode.cpp]: done

File: src/lib_gui/qt/graphics/graph/QtGraphNode.cpp (hunk 5 above)
Comment (Eng-MohamedHussien) [reply src/lib_gui/qt/graphics/graph/QtGraphNode.cpp]: done

File: src/lib_gui/qt/graphics/graph/QtGraphNode.cpp (hunk 4 above)
Comment (Eng-MohamedHussien) [reply src/lib_gui/qt/graphics/graph/QtGraphNode.cpp]: done

File: src/messaging/type/graph/MessageGraphNodeMove.h (hunk 2 above)
Comment (Eng-MohamedHussien) [reply src/messaging/type/graph/MessageGraphNodeMove.h]: done

File: src/lib_gui/qt/graphics/graph/QtGraphNode.cpp (hunk 3 above)
Comment (Eng-MohamedHussien) [reply src/lib_gui/qt/graphics/graph/QtGraphNode.cpp]: done

**Review by wow2006 — APPROVED** [review]

**Pull request merged**